lib/voicemail-admin.js

```diff
@@ -0,0 +1,779 @@
+ */
+function invalidSyntax(entryComponents) {
+  throw new Error(sprintf('Invalid Syntax for \'%s %s\'',
+    entryComponents[0], entryComponents[1]));
+}
+
+/**
+ * Returns true, prints a cute little exit message
+ */
+function exit() {
+  logger.info('So long and thanks for all the fish'.cyan);
+  return true;
+}
+
+/**
+ * Returns full suite of commands needed to defined in menu.json
+ *
+ * All commands have arguments of chosenCommand and entryComponents
+ * chosenCommand is the associative array that defines details about
+ * the command currently in execution. entryComponents is an array
+ * containing the user-inputted elements of the command on a word
+ * by word basis (though words can be joined into a single element by
+ * putting them between quotation marks).
+ */
+function createApi() {
+  return {
+    showHelp: function(chosenCommand, entryComponents) {
+      if (entryComponents.length >= 2) {
+        return showHelpCommand(entryComponents);
+      }
+
+      logger.info(sprintf('%-70s'.underline, 'Command List:'));
+      for (var key in menu.commands) {
+        var command = menu.commands[key];
+        logger.info(sprintf('%s'.green + ' - %s',
+                            key, command.description));
+      }
+    },
+
+    exit: function(chosenCommand, entryComponents) {
+      return exit();
+    },
+
+    showContexts: function(chosenCommand, entryComponents) {
+      return dal.context.all()
+        .then(function(contexts) {
+          logger.info('domain          '.underline);
+          contexts.forEach(function (context) {
+            logger.info(context.domain);
+          });
+        });
+    },
+
+    showFolders: function(chosenCommand, entryComponents) {
+      return dal.folder.all()
+        .then(function(folders) {
+
+          logger.info(sprintf('%-12s %-6s %-30s'.underline,
+                              'name', 'dtmf', 'recording'));
+          for (var key in folders) {
+            var folder = folders[key];
+
+            if (key === 'add') {
+              /* This is not a folder. */
+              continue;
+            }
+
+            logger.info(sprintf('%-12s %-6s %s',
+              folder.name, folder.dtmf, folder.recording));
+          }
+        });
+    },
+
+    showFolder: function(chosenCommand, entryComponents) {
+      if (entryComponents.length !== 3) {
+        return invalidSyntax(entryComponents);
+      }
+
+      var folderName = entryComponents[2];
+
+      return dal.folder.get(folderName)
+        .then(function(folder) {
+          if (!folder) {
+            throw new Error(sprintf('Folder \'%s\' not found.', folderName));
+          }
+
+          removeUgly(folder);
+          logger.info('%j', folder);
+        });
+    },
+
+    showMailbox: function(chosenCommand, entryComponents) {
+      if (entryComponents.length !== 3) {
+        return invalidSyntax(entryComponents);
+      }
+
+      var mailboxKey = entryComponents[2].split('@');
```

Comment:
This is another area where commander would help a lot and reduce the code necessary to validate inputs.